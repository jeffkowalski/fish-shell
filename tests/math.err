
####################
# Validate basic expressions

####################
# Validate how variables in an expression are handled

####################
# Validate math error reporting
math: Invalid expression: Too few parameters for function "min" at expression position 5
math: Invalid expression: String value used where a numerical argument is expected.
math: Invalid expression: Too few parameters for function "max" at expression position 5
math: Invalid expression: Too few parameters for function "sum" at expression position 5
