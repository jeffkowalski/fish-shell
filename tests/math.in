logmsg Validate basic expressions
math 3 / 2
math 10/6
math -s0 10 / 6
math -s3 10/6
math '10 % 6'
math -s0 '10 % 6'
math '23 % 7'
math --scale=6 '5 / 3 * 0.3'
math "7^2"
math -1 + 1
math '-2 * -2'
math 5 \* -2
math -- -4 / 2
math -- '-4 * 2'

logmsg Validate how variables in an expression are handled
math $x + 1
set x 1
math $x + 1
set x 3
set y 1.5
math "-$x * $y"
math -s1 "-$x * $y"

logmsg Validate math error reporting
not math '2 - '
not math 'ncr(1)'
not math 'max()'
not math 'sin()'
not math '2 + 2 4'
not math
not math -s 12
not math 2^650
not math 1 / 0
