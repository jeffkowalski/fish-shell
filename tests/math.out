
####################
# Validate basic expressions
1
1
1
1.667
4
4
2
0.500000
49
0
4
-10
-2
-8

####################
# Validate how variables in an expression are handled
1
2
-4
-4.5

####################
# Validate math error reporting
